src/Enum.php

```diff
@@ -121,10 +122,42 @@ public static function isValid($value) : bool
      *
      * @return array
      */
-    protected static function getDeclaredConstants() : array
+    protected static function getDeclaredConstants(): array
     {
         $reflection = new ReflectionClass(get_called_class());
 
         return (array) $reflection->getConstants();
     }
+
+    /**
+     * @return array
+     */
+    public static function toSelect(): array
+    {
+        return static::collect()
+            ->mapWithKeys(function ($key) {
+                $outcome = self::ofType($key);
+
+                return [$outcome->value() => $outcome->line()];
+            })
+            ->toArray();
+    }
+
+    /**
+     * @return string
+     */
+    public function line(): string
+    {
+        $vendor = $this->vendor() ?: null;
+
+        return Lang::get(
+            ($vendor ? $vendor.'::' : '').'enum.'.$this->langKey().'.'.str_replace('_', '-', $this->value())
+        );
+    }
+
+    abstract public function langKey(): string;
```

Comment:
Can you move this to the top of the file please between the properties and the first method.